src/webkit/wkPageProxy.ts

```diff
@@ -138,26 +109,38 @@ export class WKPageProxy {
     });
     assert(targetInfo.type === 'page', 'Only page targets are expected in WebKit, received: ' + targetInfo.type);
     this._sessions.set(targetInfo.targetId, session);
-    if (this._firstTargetCallback) {
-      this._firstTargetCallback();
-      this._firstTargetCallback = undefined;
-    }
-    if (targetInfo.isProvisional) {
-      (session as any)[isPovisionalSymbol] = true;
-      if (this._wkPage) {
-        const provisionalPageInitialized = this._wkPage.initializeProvisionalPage(session);
-        if (targetInfo.isPaused)
-          provisionalPageInitialized.then(() => this._resumeTarget(targetInfo.targetId));
-      } else if (targetInfo.isPaused) {
+
+    if (!this._initialized) {
+      assert(!targetInfo.isProvisional);
+      this._initialized = true;
+      let page: Page | undefined;
+      let error: Error | undefined;
+      try {
+        await this._wkPage.initialize(session);
+        page = this._wkPage._page;
+      } catch (e) {
+        if (!this._pageProxySession.isDisposed())
+          error = e;
+      }
+      if (error)
+        this._pagePromiseReject(error);
+      else
+        this._pagePromiseFulfill(page!);
```

Comment:
page could be undefined here if wkProxySession is disposed, should we change the type of the promise/callback?

Reply:
Good catch, updated.